### ISSUE TITLE
Porting to Python 3

### ISSUE THREAD

**Issue by pradyunsg** [osdlyrics/osdlyrics#10]
Happy New Year! Just making an issue as per the ROADMAP.md. Consider making a "0.5" github milestone and adding this.

I'll volunteer for trying to get some time in to work on porting to Python 3.

adding a ref to #8

**Comment by pradyunsg** [comment]
Do you think it's worth it to go ahead and make issues for the various items on the roadmap for triaging?

**Comment by lenoch** [comment]
@pradyunsg You are more than welcome to start a discussion in https://gitter.im/OSDLyrics/Lobby

I think it makes sense to run `2to3 --write --nobackups .` in the source directory and see what changes are helpful (I would say most of them, except `list()` in `for` statements). When I tried to build a package after running 2to3, I got the following tracebacks (when invoking /usr/bin/osdlyrics-daemon directly):

```
ERROR:dbus.proxies:Introspect error on :1.1364:/org/osdlyrics/LyricSourcePlugin/xiami: dbus.exceptions.DBusException: org.freedesktop.DBus.Python.TypeError: Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/dbus/service.py", line 707, in _message_cb
    retval = candidate_method(self, *args, **keywords)
  File "/usr/lib/python3.6/site-packages/osdlyrics/dbusext/service.py", line 207, in Introspect
    xet.tostring(node)
TypeError: must be str, not bytes
```

**Comment by pradyunsg** [comment]
I think #17 fixed this? Closing.

**Comment by Corax26** [comment]
I'm not sure this should be closed, #17 doesn't address everything needed for Python 3 to be supported.

**Comment by pradyunsg** [comment]
Oh, feel free to open a new issue then. :)
